<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Lieven Clement" />


<title>GWAS background</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>





<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GWAS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-chalkboard-teacher"></span>
     
    Journal clubs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="boltLmm.html">1. BOLT-LMM</a>
    </li>
    <li>
      <a href="regenie.html">2. REGENIE</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="https://github.com/statOmics/gwasVaudelApril2024">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="http://statomics.github.io/">statOmics</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GWAS background</h1>
<h4 class="author">Lieven Clement</h4>
<h4 class="date">statOmics, Ghent University (<a href="https://statomics.github.io" class="uri">https://statomics.github.io</a>)</h4>

</div>


<div id="background" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Background</h1>
<div id="dna" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> DNA</h2>
<ul>
<li>6 Billion base pairs: 3 billion from father and 3 billion from mother</li>
<li>Organised in <span class="math inline">\(2 \times 23\)</span> chromosomes of length 50 - 250 milion bp</li>
</ul>
<p><img src="cellChromosomeDnaBase.jpg" />
</br></p>
</div>
<div id="transcription-translation" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Transcription / translation</h2>
<p><img src="gene.svg" /></p>
</div>
<div id="variation-in-dna" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Variation in DNA</h2>
<ul>
<li><p>90% of variation in DNA are SNP: single nucleotide polymorphism. Different base at a single position in the DNA</p></li>
<li><p>Humans: <span class="math inline">\(\pm\)</span> 5 million SNPs</p></li>
<li><p>Most of them are neutral: high redundancy in the genomic code</p></li>
<li><p>Sometimes they are not neutral:</p></li>
</ul>
<p><img src="Sickle_Cell_Anemia_wiki3.png" style="width:25.0%" /><img src="sickleCellWikipedia2.png" style="width:75.0%" /></p>
<ul>
<li>Genomic recombination of parental chromosomes when producing germ cells.</li>
</ul>
<p><img src="recombination.jpeg" /></p>
<ul>
<li>Linkage disequilibrium: SNPs often occur together because of genomic recombination!</li>
</ul>
</div>
<div id="gwas-in-university-of-bergen" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> GWAS in University of Bergen</h2>
<ul>
<li>GWAS: Genome Wide Association Studies</li>
<li>Studies in large cohorts</li>
<li>Use SNPs to identify genes associated with a particular trait: e.g. birth weight, pacenta weight, BMI, … .</li>
</ul>
<p><img src="gwasContextVaudel2.png" /></p>
</div>
</div>
<div id="linear-models-for-gwas" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Linear models for GWAS</h1>
<p>In GWAS one often corrects for population stratification using the following linear model.</p>
<p><span class="math display">\[
\tag{1}
\mathbf{y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{X}_\text{PCA} \boldsymbol{\beta}_\text{PCA} +\boldsymbol{\epsilon}
\]</span>
with</p>
<ul>
<li><span class="math inline">\(\mathbf{y}\)</span> an <span class="math inline">\(N\times1\)</span> vector of the phenotype</li>
<li><span class="math inline">\(\mathbf{x}_\text{test}\)</span> an <span class="math inline">\(N\times1\)</span> vector with the genotype for the candidate SNP</li>
<li><span class="math inline">\(\beta_\text{test}\)</span> the association of candidate SNP and the phenotype</li>
<li><span class="math inline">\(\mathbf{X}_c\)</span> an <span class="math inline">\(N\times C\)</span> matrix with the covariate pattern for <span class="math inline">\(C\)</span> known covariates (vector of ones (intercept), age, gender, batch,…)</li>
<li><span class="math inline">\(\boldsymbol{\beta}_c\)</span> the <span class="math inline">\(p\times 1\)</span> vector of parameters modeling the association of the p covariates and the phenotype.</li>
<li><span class="math inline">\(\mathbf{X}_\text{PCA}\)</span> an <span class="math inline">\(N\times p\)</span> matrix with p PCs used to correct for population stratification</li>
<li><span class="math inline">\(\boldsymbol{\epsilon}\)</span> an <span class="math inline">\(N\times 1\)</span> vector with environmental residuals that are assumed to be i.i.d. <span class="math inline">\(\epsilon_i \sim N(0,\sigma_u^2)\)</span> with <span class="math inline">\(i = 1\ldots N\)</span></li>
</ul>
<p>Let <span class="math inline">\(\mathbf{Z}\)</span> be an <span class="math inline">\(N\times M\)</span> genetic relationship matrix with all <span class="math inline">\(M\)</span> normalised genotypes.
Then with the SVD we can decompose <span class="math inline">\(\mathbf{Z}\)</span></p>
<p><span class="math display">\[
\mathbf{Z} = \mathbf{U}\boldsymbol{\Delta}\mathbf{V}^T
\]</span>
Note, that the <span class="math inline">\(N\times M\)</span> matrix <span class="math inline">\(\mathbf{V}\)</span> are also the M PCs of an PCA.
So we can approximate <span class="math inline">\(\mathbf{Z}\)</span> using a truncated PCA, e.g. by using the first <span class="math inline">\(p\)</span> PCs.
<span class="math display">\[
\mathbf{Z}_p = \mathbf{U}_{p} \boldsymbol{\Delta}_p\mathbf{V}^T_p
\]</span>
with</p>
<p><span class="math display">\[
\mathbf{X}_\text{PCA} = \mathbf{U}_{p} \boldsymbol{\Delta}_p
\]</span>
the scores on the first p PCs that can be used to correct for population stratification.</p>
</div>
<div id="linear-mixed-model-for-gwas" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Linear mixed model for GWAS</h1>
<div id="specification" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Specification</h2>
<p><span class="math display">\[
\tag{1}
\mathbf{Y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{Z}_{GRM}\mathbf{u} +\boldsymbol{\epsilon}
\]</span>
with</p>
<ul>
<li><span class="math inline">\(\mathbf{Y}\)</span> an <span class="math inline">\(N\times1\)</span> vector of the phenotype</li>
<li><span class="math inline">\(\mathbf{x}_\text{test}\)</span> an <span class="math inline">\(N\times1\)</span> vector with the genotype for the candidate SNP</li>
<li><span class="math inline">\(\beta_\text{test}\)</span> the association of candidate SNP and the phenotype</li>
<li><span class="math inline">\(\mathbf{X}_c\)</span> an <span class="math inline">\(N\times p\)</span> matrix with the covariate pattern for <span class="math inline">\(C\)</span> known covariates (vector of ones (intercept), age, gender, batch,…)</li>
<li><span class="math inline">\(\boldsymbol{\beta}_c\)</span> the <span class="math inline">\(p\times 1\)</span> vector of parameters modeling the association of the p covariates and the phenotype.</li>
<li><span class="math inline">\(\mathbf{Z}\)</span> an <span class="math inline">\(N\times M\)</span> genetic relationship matrix (GRM) with all normalised genotypes</li>
<li><span class="math inline">\(\mathbf{u}\)</span> an <span class="math inline">\(M\times 1\)</span> vector with i.i.d. random effect for each SNP <span class="math inline">\(\mathbf{u}\sim \text{MVN}(0,\mathbf{I}\sigma_u^2)\)</span></li>
<li><span class="math inline">\(\boldsymbol{\epsilon}\)</span> an <span class="math inline">\(N\times 1\)</span> vector with environmental residuals that are assumed to be independent of <span class="math inline">\(\mathbf{u}\)</span> and i.i.d. <span class="math inline">\(\boldsymbol{\epsilon}\sim \text{MVN}(0,\mathbf{I}\sigma_\epsilon^2)\)</span></li>
</ul>
<p>Random effects are used to model the correlation structure in the data. They imply a certain covariance structure of <span class="math inline">\(\mathbf{y}\)</span></p>
</div>
<div id="covariance-structure" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Covariance structure</h2>
<p>Covariance structure of <span class="math inline">\(\mathbf{y}\)</span> implied by GWAS mixed model:</p>
<p><span class="math display">\[
\begin{array}{ccl}
\text{var}\left[\mathbf{Y}\right] &amp;=&amp; \text{var}\left[\mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{Z}_\text{GRM}\mathbf{u} +\boldsymbol{\epsilon}\right]\\\\
&amp;\updownarrow&amp; \mathbf{u} \perp \boldsymbol{\epsilon}\\\\
&amp;=&amp; \text{var}[\mathbf{Z}_\text{GRM}\mathbf{u}] + \text{var}[\boldsymbol{\epsilon}]\\\\
&amp;=&amp;\mathbf{Z}_\text{GRM}\text{var}[\mathbf{u}]\mathbf{Z}_\text{GRM}^T + \mathbf{I} \sigma^2\\\\
&amp;=&amp;\mathbf{Z}_\text{GRM}\mathbf{I}\sigma^2_u\mathbf{Z}_\text{GRM}^T + \mathbf{I} \sigma^2_\epsilon \\\\
&amp;=&amp;\mathbf{Z}_\text{GRM}\mathbf{Z}_\text{GRM}^T \sigma^2_u+ \mathbf{I} \sigma^2_\epsilon
\end{array}
\]</span></p>
<p>Note that the model is often also written in another way:</p>
<p><span class="math display">\[
\tag{1}
\mathbf{Y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{g} +\boldsymbol{\epsilon}
\]</span></p>
<ul>
<li><p>with <span class="math inline">\(\mathbf{g} \sim \text{MVN}(\mathbf{0},\mathbf{K}\sigma^2_g)\)</span></p></li>
<li><p><span class="math inline">\(\mathbf{K}\)</span> the <span class="math inline">\(N \times N\)</span> empirical kinship matrix</p></li>
</ul>
<p><span class="math display">\[
\mathbf{K} = \frac{\mathbf{Z}_\text{GRM}\mathbf{Z}^T_\text{GRM}}{M}
\]</span></p>
<ul>
<li><span class="math inline">\(\sigma_g^2\)</span> the polygenic variance <span class="math inline">\(\sigma_g^2=M\sigma_u^2\)</span></li>
</ul>
</div>
<div id="main-advantages-of-lmm-method" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Main advantages of LMM method</h2>
<ol style="list-style-type: decimal">
<li>Better control of false positive associations by correcting for population or relatedness structure</li>
<li>An increase in power:</li>
</ol>
<ul>
<li>Through the correction for this structure.</li>
<li>by conditioning on associated loci other than the candidate locus.</li>
</ul>
</div>
<div id="pitfalls-of-lmm" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Pitfalls of LMM</h2>
<ol style="list-style-type: decimal">
<li>Computational complexity:</li>
</ol>
<ul>
<li><span class="math inline">\(M &gt; 500.000\)</span>, <span class="math inline">\(N &gt; 70000\)</span></li>
<li>Building the GRM (<span class="math inline">\(M \times M\)</span> matrix)</li>
<li>Estimating the mean and variance components for each of the <span class="math inline">\(M\)</span> candidate SNP!</li>
<li>Association statistics for each variant (for each SNP!)</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li><p>Loss in power when the candidate marker is included in the GRM</p></li>
<li><p>Using a small subset of markers in the GRM can compromise correction for stratification</p></li>
</ol>
</div>
</div>

<div id="rmd-source-code">---
title: "GWAS background"
author: "Lieven Clement"
date: "statOmics, Ghent University (https://statomics.github.io)"
output:
    bookdown::html_document2:
      code_download: true
      df_print: paged
      theme: flatly
      highlight: tango
      toc: true
      toc_float: true
      number_sections: true
      code_folding: show
---

# Background

## DNA

- 6 Billion base pairs: 3 billion from father and 3 billion from mother
- Organised in $2 \times 23$ chromosomes of length 50 - 250 milion bp

![](cellChromosomeDnaBase.jpg)
</br>

## Transcription / translation 

![](gene.svg)


## Variation in DNA

- 90% of variation in DNA are SNP: single nucleotide polymorphism. Different base at a single position in the DNA

- Humans: $\pm$ 5 million SNPs

- Most of them are neutral: high redundancy in the genomic code

- Sometimes they are not neutral:

![](Sickle_Cell_Anemia_wiki3.png){width=25%}![](sickleCellWikipedia2.png){width=75%}




- Genomic recombination of parental chromosomes when producing germ cells. 

![](recombination.jpeg)

- Linkage disequilibrium: SNPs often occur together because of genomic recombination!

## GWAS in University of Bergen 

- GWAS: Genome Wide Association Studies
- Studies in large cohorts
- Use SNPs to identify genes associated with a particular trait: e.g. birth weight, pacenta weight, BMI, ... .

![](gwasContextVaudel2.png)


# Linear models for GWAS

In GWAS one often corrects for population stratification using the following linear model. 

$$
\tag{1}
\mathbf{y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{X}_\text{PCA} \boldsymbol{\beta}_\text{PCA} +\boldsymbol{\epsilon} 
$$
with 

- $\mathbf{y}$ an $N\times1$ vector of the phenotype
- $\mathbf{x}_\text{test}$ an $N\times1$ vector with the genotype for the candidate SNP
- $\beta_\text{test}$ the association of candidate SNP and the phenotype
- $\mathbf{X}_c$ an $N\times C$ matrix with the covariate pattern for $C$ known covariates (vector of ones (intercept), age, gender, batch,...) 
- $\boldsymbol{\beta}_c$ the $p\times 1$ vector of parameters modeling the association of the p covariates and the phenotype. 
- $\mathbf{X}_\text{PCA}$ an $N\times p$ matrix with p PCs used to correct for population stratification
- $\boldsymbol{\epsilon}$ an $N\times 1$ vector with environmental residuals that are assumed to be i.i.d.  $\epsilon_i \sim N(0,\sigma_u^2)$ with $i = 1\ldots N$


Let $\mathbf{Z}$ be an $N\times M$ genetic relationship matrix with all $M$ normalised genotypes. 
Then with the SVD we can decompose $\mathbf{Z}$

$$
\mathbf{Z} = \mathbf{U}\boldsymbol{\Delta}\mathbf{V}^T
$$
Note, that the $N\times M$ matrix $\mathbf{V}$ are also the M PCs of an PCA. 
So we can approximate $\mathbf{Z}$ using a truncated PCA, e.g. by using the first $p$ PCs. 
$$
\mathbf{Z}_p = \mathbf{U}_{p} \boldsymbol{\Delta}_p\mathbf{V}^T_p
$$
with 

$$
\mathbf{X}_\text{PCA} = \mathbf{U}_{p} \boldsymbol{\Delta}_p
$$
the scores on the first p PCs that can be used to correct for population stratification. 

# Linear mixed model for GWAS

## Specification

$$
\tag{1}
\mathbf{Y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{Z}_{GRM}\mathbf{u} +\boldsymbol{\epsilon} 
$$
with 

- $\mathbf{Y}$ an $N\times1$ vector of the phenotype
- $\mathbf{x}_\text{test}$ an $N\times1$ vector with the genotype for the candidate SNP
- $\beta_\text{test}$ the association of candidate SNP and the phenotype
- $\mathbf{X}_c$ an $N\times p$ matrix with the covariate pattern for $C$ known covariates (vector of ones (intercept), age, gender, batch,...) 
- $\boldsymbol{\beta}_c$ the $p\times 1$ vector of parameters modeling the association of the p covariates and the phenotype. 
- $\mathbf{Z}$ an $N\times M$ genetic relationship matrix (GRM) with all normalised genotypes
- $\mathbf{u}$ an $M\times 1$ vector with i.i.d. random effect for each SNP $\mathbf{u}\sim \text{MVN}(0,\mathbf{I}\sigma_u^2)$
- $\boldsymbol{\epsilon}$ an $N\times 1$ vector with environmental residuals that are assumed to be independent of $\mathbf{u}$ and i.i.d.  $\boldsymbol{\epsilon}\sim \text{MVN}(0,\mathbf{I}\sigma_\epsilon^2)$

Random effects are used to model the correlation structure in the data. They imply a certain covariance structure of $\mathbf{y}$

## Covariance structure

Covariance structure of $\mathbf{y}$ implied by GWAS mixed model: 

$$
\begin{array}{ccl}
\text{var}\left[\mathbf{Y}\right] &=& \text{var}\left[\mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{Z}_\text{GRM}\mathbf{u} +\boldsymbol{\epsilon}\right]\\\\
&\updownarrow& \mathbf{u} \perp \boldsymbol{\epsilon}\\\\
&=& \text{var}[\mathbf{Z}_\text{GRM}\mathbf{u}] + \text{var}[\boldsymbol{\epsilon}]\\\\
&=&\mathbf{Z}_\text{GRM}\text{var}[\mathbf{u}]\mathbf{Z}_\text{GRM}^T + \mathbf{I} \sigma^2\\\\
&=&\mathbf{Z}_\text{GRM}\mathbf{I}\sigma^2_u\mathbf{Z}_\text{GRM}^T + \mathbf{I} \sigma^2_\epsilon \\\\
&=&\mathbf{Z}_\text{GRM}\mathbf{Z}_\text{GRM}^T \sigma^2_u+ \mathbf{I} \sigma^2_\epsilon
\end{array}
$$

Note that the model is often also written in another way:

$$
\tag{1}
\mathbf{Y} = \mathbf{x}_\text{test}\beta_\text{test} + \mathbf{X}_c\boldsymbol{\beta}_c + \mathbf{g} +\boldsymbol{\epsilon} 
$$

- with $\mathbf{g} \sim \text{MVN}(\mathbf{0},\mathbf{K}\sigma^2_g)$ 


- $\mathbf{K}$ the $N \times N$ empirical kinship matrix 

$$
\mathbf{K} = \frac{\mathbf{Z}_\text{GRM}\mathbf{Z}^T_\text{GRM}}{M} 
$$ 

- $\sigma_g^2$ the polygenic variance $\sigma_g^2=M\sigma_u^2$

## Main advantages of LMM method

1. Better control of false positive associations by correcting for population or relatedness structure 
2. An increase in power: 
- Through the correction for this structure. 
- by conditioning on associated loci other than the candidate locus. 

## Pitfalls of LMM

1. Computational complexity: 

  - $M > 500.000$, $N > 70000$
  - Building the GRM ($M \times M$ matrix)
  - Estimating the mean and variance components  for each of the $M$ candidate SNP! 
  - Association statistics for each variant (for each SNP!)

2. Loss in power when the candidate marker is included in the GRM

3. Using a small subset of markers in the GRM can compromise correction for stratification


</div>
<div class="footer">
    <hr>
    This work is licensed under the <a href= "https://creativecommons.org/licenses/by-nc-sa/4.0">
    CC BY-NC-SA 4.0</a> licence.
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("gwasBackground.Rmd");
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
